src/mac/mac_messages.c

```diff
@@ -154,8 +154,8 @@ MacMessage mac_msg_create_ul_data_ack(uint8_t ack_type, uint8_t seqNr,
   genericmsg->payload_len = 0;
 
   genericmsg->hdr_bin[0] = (ul_data_ack & 0b111) << 5;
-  genericmsg->hdr_bin[0] |= (ack_type & 0b11) << 3;
-  genericmsg->hdr_bin[0] |= (seqNr & 0b111);
+  genericmsg->hdr_bin[0] |= (ack_type & 0b1) << 4;
+  genericmsg->hdr_bin[0] |= (seqNr & 0b1111);
   genericmsg->hdr_bin[1] = (fragNr & 0b11111) << 3;
```

Comment:
Is it possible to add a unit test for this?

Reply:
There is a unit test for dl_data_ack and ul_data_ack messages included in this branch.

test_mac_msg_parse_ul_data_ack()
test_mac_msg_parse_dl_data_ack()

in test_mac_messages.c

But I think it may be better to use the boundary values seqNr=15 and fragNr=31 for the test instead of the currently randomly chosen seqNr=5 and fragNr=13